contracts/lib/QuoteV3Auth/V3Parser.sol

```diff
@@ -154,4 +154,120 @@ library V3Parser {
 
         success = true;
     }
+
+    /// for structured quote parse
+    function validateParsedInput(V3Struct.ParsedV3QuoteStruct calldata v3Quote)
+        internal
+        pure
+        returns (
+            bool success,
+            V3Struct.Header memory header,
+            V3Struct.EnclaveReport memory localEnclaveReport,
+            bytes memory signedQuoteData, // concatenation of header and local enclave report bytes
+            V3Struct.ParsedECDSAQuoteV3AuthData memory authDataV3
+        )
+    {
+        success = true;
+        localEnclaveReport = v3Quote.localEnclaveReport;
+        V3Struct.EnclaveReport memory pckSignedQeReport = v3Quote.v3AuthData.pckSignedQeReport;
+
+        require(
+            localEnclaveReport.reserved3.length == 96 && localEnclaveReport.reserved4.length == 60
+                && localEnclaveReport.reportData.length == 64,
+            "local QE report has wrong length"
+        );
+        require(
+            pckSignedQeReport.reserved3.length == 96 && pckSignedQeReport.reserved4.length == 60
+                && pckSignedQeReport.reportData.length == 64,
+            "QE report has wrong length"
+        );
+        require(
+            v3Quote.v3AuthData.certification.certType == 5,
+            "certType must be 5: Concatenated PCK Cert Chain (PEM formatted)"
+        );
+        require(v3Quote.v3AuthData.certification.decodedCertDataArray.length == 3, "3 certs in chain");
+        require(
+            v3Quote.v3AuthData.ecdsa256BitSignature.length == 64 && v3Quote.v3AuthData.ecdsaAttestationKey.length == 64
+                && v3Quote.v3AuthData.qeReportSignature.length == 64,
+            "Invalid ECDSA signature format"
+        );
+        require(
+            v3Quote.v3AuthData.qeAuthData.parsedDataSize == v3Quote.v3AuthData.qeAuthData.data.length,
+            "Invalid QEAuthData size"
+        );
+
+        // todo!
+        // v3Quote.v3AuthData.certification.certDataSize ==
+        //      len(join((BEGIN_CERT, base64.encode(certArray[i]), END_CERT) for i in 0..3))
+        // This check need b64 encoding, skip it now.
+        // require(
+        //     base64.encode(v3Quote.v3AuthData.certification.decodedCertDataArray[0]).length +
+        //          base64.encode(v3Quote
+        //             .v3AuthData
+        //             .certification
+        //             .decodedCertDataArray[1])
+        //             .length +
+        //          base64.encode(v3Quote
+        //             .v3AuthData
+        //             .certification
+        //             .decodedCertDataArray[2])
+        //             .length +
+        //         3 *
+        //         (HEADER_LENGTH + FOOTER_LENGTH) ==
+        //         v3Quote.v3AuthData.certification.certDataSize,
+        //     "Invalid certData size"
+        // );
+
+        uint32 totalQuoteSize = 48 // header
```

Comment:
Minimum quote length is defined [here](https://github.com/intel/SGX-TDX-DCAP-QuoteVerificationLibrary/blob/16b7291a7a86e486fdfcf1dfb4be885c0cc00b4e/Src/AttestationLibrary/src/QuoteVerification/QuoteConstants.h#L95-L99)

it does not take the sizes of authData and certData values itself into account, but rather the bytes allocated to store `authDataSize` (2 bytes), `certDataSize` (4 bytes) and `certDataType` (2 bytes).

The auth_data_size_byte_len is also missing here which is 4 bytes.

Reply:
Oh, yes, missed few sizeof(DataSize)

Reply:
Consider removing this length check completely, because the numbers here adds up to exactly `MINIMUM_QUOTE_LENGTH` - 4 . 

`MINIMUM_QUOTE_LENGTH` is more appropriately used for checking quotes that are sent in raw bytes form to ensure it is conform to the proper byte length defined in the various Quote structs.

Since the input to `verifyParsedQuote()` is a Solidity struct, most values are type-safe and can safely assume to be in the correct byte length. Therefore, we only have to check for variables that are of dynamically-lengthed `bytes` type, such as `qeReport.reserved3`, `qeReport.reserved4`, ECDSA key and signature.

I am working on these changes now, and will make a staging commit that will eventually be merged into `main`.